src/main/java/org/broadinstitute/dsm/model/filter/participant/BaseFilterParticipantList.java

```diff
@@ -0,0 +1,248 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(BaseFilterParticipantList.class);
+    public static final String PARTICIPANT_DATA = "participantData";
+    public static final String OPTIONS = "OPTIONS";
+    public static final String RADIO = "RADIO";
+    public static final String PARTICIPANTS = "PARTICIPANTS";
+    protected static final Gson GSON = new Gson();
+    private static final Pattern DATE_PATTERN = Pattern.compile(
+            "^\\d{4}-\\d{2}-\\d{2}$");
+    private final ParticipantDataDao participantDataDao;
+
+    {
+        participantDataDao = new ParticipantDataDao();
+    }
+
+    public BaseFilterParticipantList() {
+        super(null);
+    }
+
+    public BaseFilterParticipantList(String jsonBody) {
+        super(jsonBody);
+    }
+
+
+    protected List<ParticipantWrapper> filterParticipantList(Filter[] filters, Map<String, DBElement> columnNameMap, @NonNull DDPInstance instance) {
+        Map<String, String> queryConditions = new HashMap<>();
+        List<ParticipantDataDto> allParticipantData = null;
+        if (filters != null && columnNameMap != null && !columnNameMap.isEmpty()) {
+            Map<String, Integer> allIdsForParticipantDataFiltering = new HashMap<>();
+            int numberOfParticipantDataFilters = 0;
+            for (Filter filter : filters) {
+                if (filter != null) {
+                    String tmp = null;
+                    if (filter.getParticipantColumn() != null) {
+                        tmp = StringUtils.isNotBlank(filter.getParentName()) ? filter.getParentName() : filter.getParticipantColumn().getTableAlias();
+                    }
+                    String tmpName = null;
+                    DBElement dbElement = null;
+                    if (filter.getFilter1() != null && StringUtils.isNotBlank(filter.getFilter1().getName())) {
+                        tmpName = filter.getFilter1().getName();
+                    } else if (filter.getFilter2() != null && StringUtils.isNotBlank(filter.getFilter2().getName())) {
+                        tmpName = filter.getFilter2().getName();
+                    }
+                    if (filter.getParticipantColumn() != null && PARTICIPANT_DATA.equals(filter.getParticipantColumn().tableAlias)) {
+                        if (allParticipantData == null) {
+                            allParticipantData = participantDataDao
+                                    .getParticipantDataByInstanceId(Integer.parseInt(instance.getDdpInstanceId()));
+                        }
+                        numberOfParticipantDataFilters++;
+                        addParticipantDataIdsForFilters(filter, tmpName, allParticipantData, allIdsForParticipantDataFiltering);
+                    } else {
+                        if (StringUtils.isNotBlank(tmpName)) {
+                            dbElement = columnNameMap.get(tmp + "." + tmpName);
```

Comment:
Could tmp be null?

Reply:
Can't say, main idea of refactoring is to keep the current behavior but change code structure to make it more organized, that's why I did not go deep details how logic is implemented, as I mentioned in my `P.S` filtering feature needs to be improved(logic) but not in refactoring part.